chain.go

```diff
@@ -263,6 +263,92 @@ func (c *Client) GetBlockHash(blockHeight int64) (*chainhash.Hash, error) {
 	return c.GetBlockHashAsync(blockHeight).Receive()
 }
 
+// FutureGetBlockHeaderResult is a future promise to deliver the result of a
+// GetBlockHeaderAsync RPC invocation (or an applicable error).
+type FutureGetBlockHeaderResult chan *response
+
+// Receive waits for the response promised by the future and returns the
+// serialized blockheader requested from the server given its hash.
+func (r FutureGetBlockHeaderResult) Receive() (string, error) {
```

Comment:
This should return a `*wire.BlockHeader` in my opinion so it's consistent with the other block-relatedAPIs which return concrete types as opposed to the raw JSON types (the hex in this case).

See the `func (r FutureGetBlockResult) Receive() (*btcutil.Block, error) {` for an example and details on how it deserializes.